include/sound/sof/topology.h

```diff
@@ -39,6 +39,8 @@ enum sof_comp_type {
 	SOF_COMP_ASRC,		/**< Asynchronous sample rate converter */
 	SOF_COMP_DCBLOCK,
 	SOF_COMP_SMART_AMP,             /**< smart amplifier component */
+
+	SOF_COMP_PROCESS = 1000,	/**< generic process component */
 	/* keep FILEREAD/FILEWRITE as the last ones */
 	SOF_COMP_FILEREAD = 10000,	/**< host test based file IO */
 	SOF_COMP_FILEWRITE = 10001,	/**< host test based file IO */
```

Comment:
@keyonjie can we enhance the commit message to specify WHY this change is needed. I'm not really sure but I think it has something to do with the effort of adding new components which are simpler enough to be abstracted to just ONE processing function.

Reply:
Maybe I can add some message from the PR to one of the commit for this. 
@dbaluta the corresponding FW PR is provided, we might have more refinement (e.g. share common codes among different process component) later but now all existed process components should work with these 2 PRs already.

Reply:
Ok, sounds good!